Support no sharding config

zhaojuanmao opened the pull request:
supporting no sharding config to make it similar to DDP algorithm

Commit:
[FSDP] Validate exec order using `compute_device`

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76664

Approved by: https://github.com/zhaojuanmao

Commit:
[FSDP] Faster dict inversion

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76665

Approved by: https://github.com/zhaojuanmao

zhaojuanmao committed:
rebase

zhaojuanmao committed:
rebase

zhaojuanmao committed:
merge conflict

zhaojuanmao committed:
merge conflict

mrshenli commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -103,25 +103,18 @@ class ShardingStrategy(Enum):
-                              among ranks, all_reduce is inserted after backward computation
-                              is done for synchronizing gradients. Full optimizer states
-                              are updated in each rank.
+    NO_SHARD: This is similar to PyTorch `DistributedDataParallel` API. Parameters, gradients
+              and optimizer states are replicated among ranks, all_reduce is inserted after
```
nit: same as on the previous PR. Double quote `DistributedDataParallel` and `all_reduce` to display as code.

mrshenli reviewed (approved):
Reviewed on #76628, stamp to unblock. 

Please remove changes in `ideep` submodule before landing.

zhaojuanmao commented:
> Reviewed on #76628, stamp to unblock.
> 
> Please remove changes in `ideep` submodule before landing.

how to remove changes in `ideep` submodule..

Commit:
fix where type promotion

Fixes #73298
I don't know whether `where` kernel actually supports type promotion, nor am I in the mood to find out, so it's manual type promotion.
Edit: nah, i can't tell TI to "promote to common dtype" because of bool condition, so manual type promotion is our only option.
I'll see what tests start failing and fix.
Uses some parts from #62084
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76691
Approved by: https://github.com/mruberry

vkuzo committed:
quant: rename _ObserverBase to UniformQuantizationObserverBase (#76461)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76461

Renaming as the old name was confusing. The name represents
better what this class is doing.

Test Plan: CI

Reviewed By: jerryzh168

Differential Revision: D35976350

Pulled By: vkuzo

fbshipit-source-id: 6da6c1767cec729c3959b13ae9dd939d0b2f622c
(cherry picked from commit 065608ef42c599525bfad4603af74c5bdf0881c3)

vkuzo committed:
quant: make RecordingObserver inherit from ObserverBase (#76460)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76460

`RecordingObserver` inherits from `_ObserverBase` but does not use any functionality
from it. Making it inherit from `ObserverBase` instead.

This will make it simpler to rename `_ObserverBase` to something more meaningful in the next PR.

Test Plan: CI

Reviewed By: jerryzh168

Differential Revision: D35976351

Pulled By: vkuzo

fbshipit-source-id: 19c106bf0d48607c231702e2e048f42a7f48a5c6
(cherry picked from commit 4fd44123b0e9bcdcae546aecabe80d7642129cf5)

vkuzo committed:
ns for fx: more meaningful error message when creating shadow model (#76468)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76468

This makes the error message when copying an unsupported node more verbose.
This is useful to debug where specifically in a user model this is failing.

Test Plan:
1. hardcode this condition to hit
2. run NS tests
3. verify the exception now prints details about the offending node

Reviewed By: jerryzh168

Differential Revision: D35978652

Pulled By: vkuzo

fbshipit-source-id: 9cc93dfa46469bf6ef60aa38d4011041b6709df9
(cherry picked from commit c6e382c2a69aba6ba66740f238bc14446521a433)

vkuzo committed:
ns for fx: skip operator.add and operator.mul when shadowing (#76504)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76504

Shadowing for add and mul is not implemented, this PR fixes the skipping
logic to also skip the `operator.add` and `operator.mul` flavor of these
operators.

Test Plan:
```
python test/test_quantization.py -k test_mul_add_skips_shadowing
```

Reviewed By: dzdang

Differential Revision: D35985997

Pulled By: vkuzo

fbshipit-source-id: f832e54a5461d3b182df4bb905357d6c66742e98
(cherry picked from commit 93ae9592f68873865ebfdc438bffb1c9486dd1c1)

wonjoo-wj committed:
Remove deprecated string torch::lazy::BackendDevice constructor (#76506)

Summary:
Remove deprecated string torch::lazy::BackendDevice constructor, re-landing part of https://github.com/pytorch/pytorch/pull/76264.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76506

Reviewed By: dagitses

Differential Revision: D35993059

Pulled By: alanwaketan

fbshipit-source-id: c859331919447ecfa56a9a57a3324305b7904fc2
(cherry picked from commit 2e99b1533425693a6575405f8806551efb2002e4)

Commit:
move generate-code into shared build structure (#75699)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/75699

ghstack-source-id: 155255334

Test Plan: Rely on CI.

Reviewed By: dreiss

Differential Revision: D35587412

fbshipit-source-id: 5ab79c07029de279a1fae36519654a73bb61d430
(cherry picked from commit 4896b72a6c0cc087e36889d21d2d885009d94a6d)

awfeequdng committed:
check parameter k and l

Fixes #76715

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76719
Approved by: https://github.com/ezyang

Commit:
disable the //:generate-code target in Bazel (#76174)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76174

This is about to conflict with the existing Bazel codegen
outputs. Switch to it atomically.
ghstack-source-id: 155029309

Test Plan: Verify manually and rely on CI.

Reviewed By: dreiss

Differential Revision: D35815288

fbshipit-source-id: 8b35e7baeb8572aef13c07cac689ee84dc7335d5
(cherry picked from commit 6dde9831a30fcf664b73fccaa51e30a7049b3251)

Commit:
add support for filtering out Bazel targets from common structure (#76173)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/76173

We need this facility temporarily to sequence some changes without
breakage. This is generally not a good idea since the main purpose of
this effort is to replicate builds in OSS Bazel.
ghstack-source-id: 155215491

Test Plan: Manual test and rely on CI.

Reviewed By: dreiss

Differential Revision: D35815290

fbshipit-source-id: 89bacda373e7ba03d6a3fcbcaa5af42ae5eac154
(cherry picked from commit 1b808bbc94c939da1fd410d81b22d43bdfe1cda0)

zhaojuanmao committed:
Revert "thirdparty"changes

This reverts commit ef13f2d1fa3c88122d1cc80edad1a0a18dc093b3.

zhaojuanmao committed:
Revert "thirdparty"changes

This reverts commit ef13f2d1fa3c88122d1cc80edad1a0a18dc093b3.

zhaojuanmao commented:
> > Reviewed on #76628, stamp to unblock.
> > Please remove changes in `ideep` submodule before landing.
> 
> how to remove changes in `ideep` submodule..

removed

zhaojuanmao committed:
address docs

zhaojuanmao committed:
Merge branch 'noShard' of https://github.com/pytorch/pytorch into noShard

zhaojuanmao committed:
Revert "address docs"

This reverts commit 351c5c47d3feb1a267c564bc4b4e72602390fb84.

zhaojuanmao commented:
the commits messed up:(, closing this, will create a new PR
